### ISSUE TITLE
Smart spacing on the `dif` symbol

### ISSUE THREAD

**Issue by yaksher** [typst/typst#2515]
### Description

Currently, the `dif` symbol appears to simply have fixed spacing before the upright `d` character whenever it is preceded by anything. It would be nice if certain preceding characters would collapse the spacing, such as `(dif x)` rendering without a space before `dif`. I am assuming this should be relatively simple since it already omits the spacing when preceded by nothing, as in `dif/(dif x)`.

### Use Case

The syntax `integral_S f(x) mu (dif x)` is common in measure and probability theory and it would be nice to render the symbol without an awkward space after the parenthesis. Additionally, physics and differential topology both frequently treat differentials as normal objects and thus would run into similar issues, though I haven't done much of either since switching to Typst.

**Comment by Trebor-Huang** [comment]
For an example in differential geometry, we have expressions like $\phi^* (\mathrm{d}x)$ denoting the pullback of differential forms.

**Comment by LU15W1R7H** [comment]
The same applies to Lebesgue-integrals in measure theory.
 $integral_A f(x) mu(dif x)$

**Comment by YDX-2147483647** [comment]
I've been putting up with it for a long time… Before #2950 is merged and released, we have the following workarounds.

`#repr(math.dif)` unravels the magic:

```typst
[(
  h(amount: 0.16666666666666666em, weak: true),
  math-style(body: [d], italic: false),
)]
```

So `upright(d)` just works.

Besides, [the physica package offers a dedicated `dd(…)`](https://github.com/Leedehai/typst-physics/blob/94c89824cbb69d1513e49ea7ea3d0bd90f7016fd/physica.typ#L438-L482).

```typst
#import "@preview/physica:0.9.0": dd

$ integral abs(dd(z)) $
```

**Comment by laurmaedje** [comment]
Fixed by #2950 unless I'm mistaken.
